File: todo-list/src/Item.java
import java.time.LocalDate;

public class Item {
    // ATTRIBUTES
    private int id;
    private String description;
    private boolean done;
    private LocalDate date;

    // NEW ITEM CONSTRUCTOR
    public Item(String description){
        this.description = description;
        try{
            if(description != null){
                done = false;
                date = LocalDate.now();
            }
            else{
                throw new Exception("DESCRIPTION CANNOT BE EMPTY.");
            }
        } catch (Exception e){
            System.out.println(e.getMessage());
        }
    }

    // CONSTRUCTOR FOR SQL QUERY
    public Item(int id, String description, boolean done, LocalDate date){
        this.id = id;
        this.description = description;
        this.done = done;
        this.date = date;
    }
    public int getId() { return id; }
    public String getDescription() { return description; }
    public boolean isDone() { return done; }
    public LocalDate getDate() { return date; }
}

File: todo-list-py/src/item.py
from datetime import datetime


class Item:
    # ATTRIBUTES
    def __init__(self, description=None, id=None, done=False, date=None):
        # NEW ITEM CONSTRUCTOR
        if description is not None:
            self.description = description
            self.done = done
            self.date = date if date is not None else datetime.now().date()
            self.id = id
        else:
            raise ValueError("DESCRIPTION CANNOT BE EMPTY.")

    # METHODS
    def get_id(self):
        return self.id

    def get_description(self):
        return self.description

    def is_done(self):
        return self.done

    def get_date(self):
        return self.date
